semantic-reranker/src/main/java/org/opensearch/search/relevance/client/KendraHttpClient.java

```diff
@@ -0,0 +1,93 @@
+import com.amazonaws.http.HttpResponseHandler;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Map;
+
+import org.opensearch.search.relevance.constants.Constants;
+import org.opensearch.search.relevance.model.dto.RerankRequest;
+import org.opensearch.search.relevance.model.dto.RerankResult;
+
+public class KendraHttpClient {
+  private final AmazonHttpClient amazonHttpClient;
+  private final HttpResponseHandler<AmazonServiceException> errorHandler;
+  private final HttpResponseHandler<String> responseHandler;
+  private final AWSCredentials awsCredentials;
+  private final AWS4Signer aws4Signer;
+  private final String serviceEndpoint;
+  private final String endpointId;
+
+  private final ObjectMapper objectMapper;
+
+  public KendraHttpClient(KendraClientSettings clientSettings) {
+    // TODO: remove below line once DNS is fixed
+    AccessController.doPrivileged((PrivilegedAction<String>) () -> System.setProperty(SDKGlobalConfiguration.DISABLE_CERT_CHECKING_SYSTEM_PROPERTY, "true"));
+    
+    amazonHttpClient = AccessController.doPrivileged((PrivilegedAction<AmazonHttpClient>) () -> new AmazonHttpClient(new ClientConfiguration()));
+    errorHandler = new SimpleAwsErrorHandler();
+    responseHandler = new SimpleResponseHandler();
+    aws4Signer = new AWS4Signer();
+    aws4Signer.setServiceName(Constants.KENDRA_RANKING_SERVICE_NAME);
+    aws4Signer.setRegionName(clientSettings.getServiceRegion());
+    objectMapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    awsCredentials = clientSettings.getCredentials();
+    serviceEndpoint = clientSettings.getServiceEndpoint();
+    endpointId = clientSettings.getEndpointId();
+  }
+
+  public RerankResult rerank(RerankRequest rerankRequest) {
+    return AccessController.doPrivileged((PrivilegedAction<RerankResult>) () -> {
+      try {
+        rerankRequest.setRerankingEndpointId(endpointId);
+        System.out.println(rerankRequest.getRerankingEndpointId());
+        System.out.println(rerankRequest.getDocuments());
+        System.out.println(rerankRequest.getQueryText());
+        Request<Void> request = new DefaultRequest<>(aws4Signer.getServiceName());
+        request.setHttpMethod(HttpMethodName.POST);
+        request.setEndpoint(URI.create(serviceEndpoint));
+        request.setHeaders(Map.of("Content-Type", "application/x-amz-json-1.0", "X-Amz-Target", "AWSKendraRerankingFrontendService.Rerank"));
+        System.out.println("objectMapper: " + objectMapper.writeValueAsString(rerankRequest));
+        request.setContent(new ByteArrayInputStream(objectMapper.writeValueAsString(rerankRequest).getBytes(StandardCharsets.UTF_8)));
```

Comment:
There will be a dedicated Kendra reranking client in the AWS SDK, right?

I'm assuming this code (and the whole `dto` package) are temporary until you have added the client to the SDK.

Reply:
Yes, the service is currently under development so we don't have an external SDK. We will update this code once it is ready.